core/src/test/java/io/questdb/test/griffin/O3FailureTest.java

```diff
@@ -290,12 +289,16 @@ public long length(int fd) {
             @Override
             public int openRW(LPSZ name, long opts) {
                 int fd = TestFilesFacadeImpl.INSTANCE.openRW(name, opts);
-                if (theFd >= 0 && fd > 0 && Chars.endsWith(name, fileName)) {
-                    theFd = fd;
+                if (this.fd >= 0 && fd > 0 && Chars.endsWith(name, fileName)) {
+                    this.fd = fd;
                     return fd;
                 }
                 return fd;
             }
+
+            {
+                this.fd = 0;
```

Comment:
Why it has to be set to 0 and -1 is not good enough?

Reply:
I didn't dig too deep but test flow is a bit more complex, differentiates between these values and fails when only -1 is used.